strawberryfields/engine.py

```diff
@@ -182,58 +285,34 @@ def _run_program_locally(self, prog, **kwargs):
                 try:
```

Comment:
With a program validation stage, this step should never happen. I think we should remove it so that our tests alert us to validation errors rather than fall back to some implicity decomposition

Reply:
:+1: thumb

Reply:
Agreed.